dask/dataframe/tests/test_groupby.py

```diff
@@ -2695,20 +2710,35 @@ def test_groupby_aggregate_categoricals(grouping, agg):
     )
     ddf = dd.from_pandas(pdf, 2)
 
-    if PANDAS_GT_200:
-        ctx = pytest.raises(NotImplementedError, match="numeric_only=False")
-    else:
-        ctx = contextlib.nullcontext()
-
     # DataFrameGroupBy
-    expected = agg(grouping(pdf))
-    with ctx:
+    with check_observed_deprecation():
+        expected = agg(grouping(pdf))
+
+    observed_ctx = (
+        pytest.warns(FutureWarning, match="observed")
+        if PANDAS_GT_210
+        else contextlib.nullcontext()
+    )
+    numeric_ctx = (
+        pytest.raises(NotImplementedError, match="numeric_only=False")
+        if PANDAS_GT_200
+        else contextlib.nullcontext()
+    )
+    with observed_ctx, numeric_ctx:
```

Comment:
This ordering is totally fine, just wanted to point out as an FYI that the ordering of `pytest.warns` and `pytest.raises` matters https://github.com/pytest-dev/pytest/issues/9036